text/0040-pagination-convention.md

```diff
@@ -0,0 +1,47 @@
+- Title: Pagination Convention
+- Start Date: 2021-05-10
+- Specification PR: [#40](https://github.com/meilisearch/specifications/pull/40)
+- MeiliSearch Tracking-Issues:
+
+# Pagination Convention
+
+## 1. Feature Description and Interaction
+
+### I. Summary
+
+Pagination allows the content to be viewed in the form of a sequence in order to facilitate the browsing of its content by not overloading the number of elements to be viewed for the user. Pagination also has the advantage of easily providing the total number of items that remain to be traversed by not overloading the network with all the content that can be loaded.
+
+### II. Motivation
+
+This specification will be used as a convention to implement simple and standard paginations to be set up on API routes such as the list of updates for example. Since the pagination of search results is intrinsically linked to technical implementations of the search engine, this specific case is detailed in its own specification.
```

Comment:
Is it here we should note the different types of pagination we could expect?
- infinite scroll: no need to know if the next page exists -> only `offset` and `limit` are useful in their "natural" usage
- "infinite" pagination: the users can access the next page and are aware of the existence of this page in front end. The developer needs to check if a next page exists by using `limit + 1`
- "finite" pagination : the users can access the next page and the last page. `forceExhaustivity` needs to be used.

Reply:
The SDK should provide these 3 ways of pagination as smoothly as possible

Reply:
`forceExhaustivity` is only available on the `search` endpoint because of the internal data structures and algorithms of the engine. The current specification is intended for routes that always return a finite number of documents and where we can compute a `total` easily. All type of pagination like infinite, page based can be handled by that convention.